src/client/graphics/layers/TeamStats.ts

```diff
@@ -61,12 +66,20 @@ export class TeamStats extends LitElement implements Layer {
         let totalGold = 0n;
         let totalTroops = 0;
         let totalScoreSort = 0;
+        let totalSAMs = 0;
+        let totalLaunchers = 0;
+        let totalWarShips = 0;
+        let totalCities = 0;
 
         for (const p of teamPlayers) {
           if (p.isAlive()) {
             totalTroops += p.troops();
             totalGold += p.gold();
             totalScoreSort += p.numTilesOwned();
+            totalLaunchers += p.units(UnitType.MissileSilo).length;
+            totalSAMs += p.units(UnitType.SAMLauncher).length;
+            totalWarShips += p.units(UnitType.Warship).length;
+            totalCities += p.units(UnitType.City).length;
```

Comment:
This causes unnecessary allocations.

```suggestion
            totalLaunchers += p.totalUnitLevels(UnitType.MissileSilo);
            totalSAMs += p.totalUnitLevels(UnitType.SAMLauncher);
            totalWarShips += p.totalUnitLevels(UnitType.Warship);
            totalCities += p.totalUnitLevels(UnitType.City);
```